test/conf/cassandra.yaml

```diff
@@ -64,3 +64,28 @@ track_warnings:
     row_index_size:
         warn_threshold_kb: 4096
         abort_threshold_kb: 8192
+memtable_configurations:
+    skiplist:
+        extends: default
+        class: SkipListMemtable
+    skiplist_remapped:
+        extends: skiplist
+    test_fullname:
+        class: org.apache.cassandra.db.memtable.TestMemtable
+    test_shortname:
+        class: TestMemtable
+        skiplist: "true"
+    test_empty_class:
+        class: ""
+    test_unknown_class:
+        class: NotExisting
+    test_invalid_param:
+        class: SkipListMemtable
+        invalid: throw
+    test_invalid_extra_param:
+        extends: test_shortname
+        invalid: throw
+    test_invalid_factory_method:
+        class: org.apache.cassandra.cql3.validation.operations.CreateTest$InvalidMemtableFactoryMethod
+    test_invalid_factory_field:
+        class: org.apache.cassandra.cql3.validation.operations.CreateTest$InvalidMemtableFactoryField
```

Comment:
With [CASSANDRA-17292](https://issues.apache.org/jira/browse/CASSANDRA-17292) on the horizon, it might be a good time to think about using a structure for memtable configuration that would be compatible w/ the main proposal there: https://github.com/maedhroz/cassandra/commit/450b920e0ac072cec635e0ebcb63538ee7f1fc5a

Given we're moving toward implementation-specific configuration, we could even replace the existing ungrouped memtable YAML items with a new top-level `memtable` element. (There are a few ways to handle compatibility, including just using the items under the new `memtable` element if one is actually specified, pulling values from the old top-level options if they aren't specified in the new format, etc.)

ex.

```
memtable:
  configuration: skiplist
  configurations:
    skiplist:
      class: SkipListMemtable
    trie:
      class: TrieMemtable
      shards: 16
```

Reply:
Looks good, thanks!

Just a couple minor things to cleanup, like `CLASS_OPTION` and `EXTENDS_OPTION` become unused in `MemtableParams`.

Reply:
Removed the two properties and added format documentation.